File: 백준/Silver/10799. 쇠막대기/쇠막대기.py
n = input()
stk=[]
cnt = 0
for i in range(len(n)):
    if n[i] == '(':
        stk.append("(")
    else:
        if n[i-1] == "(":
            stk.pop()
            cnt += len(stk)
        else:
            stk.pop()
            cnt+=1
print(cnt)

File: 백준/Silver/10799. 쇠막대기/쇠막대기.cc
#include <iostream>
#include <stack>

using namespace std;


int main() {
    ios::sync_with_stdio(false), cin.tie(NULL);
    int cnt = 0;
    string inp;
    cin >> inp;
    stack<char> s;
    for (int i = 0; i < inp.length(); i++) {
        if (inp[i] == '(') {
            s.push(inp[i]);
        } else {
            if (inp[i - 1] == '(') {
                s.pop();
                cnt += s.size();
            } else {
                s.pop();
                cnt += 1;
            }
        }
    }
    cout << cnt;
}
